pkg/controller/networkpolicy/networkpolicy_controller.go

```diff
@@ -566,6 +568,12 @@ func (n *NetworkPolicyController) processNetworkPolicy(np *networkingv1.NetworkP
 	appliedToGroupKey := n.createAppliedToGroup(np.Namespace, &np.Spec.PodSelector, nil, nil)
 	appliedToGroupNames := []string{appliedToGroupKey}
 	rules := make([]controlplane.NetworkPolicyRule, 0, len(np.Spec.Ingress)+len(np.Spec.Egress))
+	// Retrieve Namespace logging annotation.
+	enableLogging := false
+	namespace, err := n.namespaceLister.Get(np.Namespace)
```

Comment:
What if the namespace annotation changes? Should it watch namespace events and trigger KNP reprocessing?

Reply:
Yes, this was the intention from requirement analysis, but it seems the current implementation only checks the 

value of an annotation when implementing network policies, so updates to namespace's annotation would not affect logging behaviour for existing network policy.

@qiyueyao is that intentional?

Reply:
It is not intentional, I believe it is important to trigger reprocessing. Updating this.